plugin/core/sessions.py

```diff
@@ -469,23 +470,62 @@ def can_start(cls, window: sublime.Window, initiating_view: sublime.View,
         """
         return None
 
+    @classmethod
+    def on_pre_start(cls, window: sublime.Window, initiating_view: sublime.View,
+                     workspace_folders: List[WorkspaceFolder], resolved: ResolvedStartupConfig) -> Optional[str]:
+        """
+        Callback invoked just before the language server subprocess is started. This is the place to do last-minute
+        adjustments to your "command" or "initializationOptions" in the passed-in "resolved" argument. You can also
+        choose to return a custom working directory, but consider that a language server shouldn't care about the
+        working directory.
+
+        :param      window:             The window
+        :param      initiating_view:    The initiating view
+        :param      workspace_folders:  The workspace folders
+        :param      resolved:           The resolved configuration
+
+        :returns:   A desired working directory, or None if you don't care
+        """
+        return None
+
+    @classmethod
+    def on_post_start(cls, window: sublime.Window, initiating_view: sublime.View,
+                      workspace_folders: List[WorkspaceFolder], resolved: ResolvedStartupConfig) -> None:
+        """
+        Callback invoked when the subprocess was just started.
+
+        :param      window:             The window
+        :param      initiating_view:    The initiating view
+        :param      workspace_folders:  The workspace folders
+        :param      resolved:           The resolved configuration
+        """
+        return None
+
     def __init__(self, weaksession: 'weakref.ref[Session]') -> None:
         """
-        Constructs a new instance.
+        Constructs a new instance. Your instance is constructed after a response to the initialize request.
 
         :param      weaksession:  A weak reference to the Session. You can grab a strong reference through
                                   self.weaksession(), but don't hold on to that reference.
         """
         self.weaksession = weaksession
 
+    def on_pre_window_settings(self, settings: DottedDict) -> None:
```

Comment:
We have `on_workspace_configuration` which follows the LSP naming and you are adding one that completely doesn't match LSP naming. Can we be consistent with the existing method and call it `on_pre_workspace_configuration` or similar? For plugin authors, it might be clearer to follow the LSP naming as they should be familiar with that. This is not made for normal ST users.